datashuttle/datashuttle.py

```diff
@@ -298,7 +308,7 @@ def write_public_key(self, filepath: str):
     def make_config_file(
```

Comment:
The arguments that default to `None` should include `None` as a possible type, otherwise `mypy` complains with:
> error: Incompatible default for argument "{ARG_NAME}" (default has type "None", argument has type "str")

Reply:
That's strange mine does not complain, are you running by any change with ```--no-implicit-optional```? If I understand [this post](https://stackoverflow.com/questions/51055952/does-a-default-parameters-overwrite-type-hints-for-mypy) correctly it seems the default behaviour should be to implicitly add None as an optional type (unless --no-implicit-optional is passed). I vaguely remember looking into this argument before but I can't see it in the config files but may have forgotton it somewhere.

Good point and I think worth discussing as a team what our preffered approach is. Mine would be to keep as default to avoid clutter, but I see some people (e.g. the person in that thread) like things stricter so worth discussing.

Reply:
Thar's strange. I simply ran `mypy -p datashuttle`. But perhaps our `mypy` installs are somehow configured differently somewhere.
I agree it's a good idea to discuss it as a team. Personally, I would prefer to avoid the clutter if possible

Reply:
I tested the `mypy` thing on a different machine and it finds no problems now

Reply:
That's strange! I will keep an eye out for this problem in the automation tests. Thanks a lot for looking into this